tests/unit/test_manof.py

```diff
@@ -0,0 +1,83 @@
+import mock
+
+from twisted.internet import defer
+from twisted.trial import unittest
+
+import manof.image
+import clients.logging
+
+logger = clients.logging.TestingClient('manof_unit').logger
+
+
+class ManofUnitTestCase(unittest.TestCase):
+
+    def __init__(self, *args, **kwargs):
+        super(ManofUnitTestCase, self).__init__(*args, **kwargs)
+
+    def setUp(self):
+        self._logger = logger
+        self._logger.info('Setting up unit test')
+
+    @property
+    def name(self):
+        return self.__class__.__name__
+
+    @defer.inlineCallbacks
+    def test_provision_pull(self):
+        self._logger.info('Testing manof provision with pull')
+        image = self._create_manof_image(
+            image_properties={
+                'image_name': 'test_image',
+            },
+            image_args={
+                'repository': None,
+                'tag_local': None,
+            })
+        yield self._assert_image_provision_commands(image, [
```

Comment:
Lets keep it even simple, the scenario here is, when manof provisioning a target, it checks if it has a context
if it doesn't, it simply use `pull`
so, I would patch manof `pull` with `mock.Mock()` and once `manof.provision()` is called, you assert that `manof.pull.assert_called_once` and that is it, you ensured that `pull` is executed when no context is given, simply as that

The alternative is, when context is given, you ensure `pull` was not called but perhaps `_run_command` with `docker build` with the argument of `context`

Also, try to keep the test very simple and tell a story